### PR TITLE
[RLlib] Apex-DDPG TrainerConfig objects.

### PR DESCRIPTION
**Pull request by sven1977**
Apex-DDPG TrainerConfig objects.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/agents/dqn/apex.py [hunk 1]
```diff
@@ -278,10 +278,28 @@ def training(
...
                 zero, there is still a chance of drawing the sample.
-            max_requests_in_flight_per_sampler_worker: Level of queuing for sampling
-                operations.
-            max_requests_in_flight_per_replay_worker: Level of queuing for replay
-                aggregator operations (if using aggregator workers).
+            max_requests_in_flight_per_sampler_worker: Max number of inflight requests
+                to each sampling worker. See the AsyncRequestsManager class for more
+                details. Tuning these values is important when running experimens with
+                large sample batches, where there is the risk that the object store may
+                fill up, causing spilling of objects to disk. This can cause any
```
Comment (maxpumperla) [inline rllib/agents/dqn/apex.py:285]: +1

File: rllib/algorithms/ddpg/apex.py [hunk 2]
```diff
@@ -7,29 +7,91 @@
...
-        "exploration_config": {"type": "PerWorkerOrnsteinUhlenbeckNoise"},
-        "n_step": 3,
-        "num_gpus": 0,
-        "num_workers": 32,
-        "replay_buffer_config": {
+        }
+        # max number of inflight requests to each sampling worker
+        # see the AsyncRequestsManager class for more details
+        # Tuning these values is important when running experimens with large sample
+        # batches. If the sample batches are large in size, then there is the risk that
```
Comment (maxpumperla) [inline rllib/algorithms/ddpg/apex.py:62]: do we need to have this here? should be covered below, right?

**Review by maxpumperla — APPROVED** [review]
just a tiny nit/question

File: rllib/algorithms/ddpg/apex.py (hunk 2 above)
Comment (sven1977) [reply rllib/algorithms/ddpg/apex.py:62]: True!

File: rllib/algorithms/ddpg/apex.py [hunk 3]
```diff
@@ -7,29 +7,91 @@
...
+        # the object store may fill up, causing the store to spill objects to disk.
+        # This can cause any asynchronous requests to become very slow, making your
+        # experiment run slowly. You can inspect the object store during your
+        # experiment via a call to ray memory on your headnode, and by using the ray
+        # dashboard. If you're seeing that the object store is filling up, turn down
+        # the number of remote requests in flight, or enable compression in your
+        # experiment of timesteps.
```
Comment (sven1977) [inline rllib/algorithms/ddpg/apex.py:69]:
```suggestion
```

File: rllib/algorithms/ddpg/apex.py (hunk 2 above)
Comment (sven1977) [reply rllib/algorithms/ddpg/apex.py]: removed.

**Pull request merged**